File: Antheneum/Antheneum.Server/Domain/Entities/EventModel.cs
namespace Domain.Entities;

public class EventModel
{
    public int EventId { get; init; }
    public int BranchId { get; init; }
    public string Title { get; init; } = string.Empty;
    public string? Description { get; init; }
    public DateTime StartDate { get; init; }
    public DateTime EndDate { get; init; }
    public int MaxSeats { get; init; }
    public string? CoverImageUrl { get; init; }
}

File: Antheneum/antheneum.client/src/app/models/events/event.model.ts
export interface Event {
  eventId: number;
  title: string;
  description: string | null;
  startDate: string;
  endDate: string;
  branchName: string;
  maxSeats: number;
  enrolledCount: number;
  coverImageUrl: string | null;
}

export interface EventDetail extends Event {
  branchId: number;
  isEnrolled: boolean;
}

export interface EventFormValue {
  eventId: number | null;
  title: string;
  description: string | null;
  branchId: number;
  startDate: string;
  endDate: string;
  maxSeats: number;
  coverImageUrl: string | null;
}

export interface EventAttendee {
  readerId: number;
  username: string;
  email: string;
  libraryCardNumber: string;
}
